blueflood-core/src/main/java/com/rackspacecloud/blueflood/inputs/formats/AggregatedPayload.java

```diff
@@ -74,6 +75,52 @@ public String toString() {
         return errors;
     }
 
+    public boolean hasDelayedMetrics(long ingestTime) {
+        if ( getDelayTime(ingestTime) > TRACKER_DELAYED_METRICS_MILLIS ) {
+            return true;
+        } else {
+            return false;
+        }
+    }
+
+    public long getDelayTime(long ingestTime) {
+        return ingestTime - timestamp;
+    }
+
+    public List<String> getAllMetricNames() {
+        List<String> metricNames = new java.util.ArrayList<String>();
+        if ( gauges != null && gauges.length > 0) {
```

Comment:
I wonder if we should start thinking functionally for some of this stuff since we can now use java 8.  No need to change.